jcl/src/java.base/share/classes/java/lang/Class.java

```diff
@@ -442,22 +442,14 @@ private static Class<?> forName(String className, Class<?> caller) throws ClassN
 		sm = System.getSecurityManager();
 	}
 	ClassLoader callerClassLoader = null;
-	if (null == sm) {
-		if (null != caller) {
-			callerClassLoader = caller.internalGetClassLoader();
-		/*[IF JAVA_SPEC_VERSION >= 19]*/
-		} else {
-			callerClassLoader = ClassLoader.internalGetSystemClassLoader();
-		/*[ENDIF] JAVA_SPEC_VERSION >= 19 */
-		}
-		return forNameImpl(className, true, callerClassLoader);
-	}
 	if (null != caller) {
-		callerClassLoader = caller.getClassLoaderImpl();
+		callerClassLoader = caller.internalGetClassLoader();
 	/*[IF JAVA_SPEC_VERSION >= 19]*/
 	} else {
 		callerClassLoader = ClassLoader.internalGetSystemClassLoader();
 	/*[ENDIF] JAVA_SPEC_VERSION >= 19 */
```

Comment:
This is missing a closing `}`. I suggest the following which also avoids the redundant assignment to `callerClassLoader` for Java 19:
```
	ClassLoader callerClassLoader;
	if (null != caller) {
		callerClassLoader = caller.internalGetClassLoader();
	} else {
		/*[IF JAVA_SPEC_VERSION >= 19]*/
		callerClassLoader = ClassLoader.internalGetSystemClassLoader();
		/*[ELSE] JAVA_SPEC_VERSION >= 19 */
		callerClassLoader = null;
		/*[ENDIF] JAVA_SPEC_VERSION >= 19 */
	}
```

Reply:
Updated.